app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -106,13 +106,13 @@ def destroy_has_many_association
       main_object.public_send("#{reflection.name.to_s.singularize}_ids").each do |associated_object_id|
         associated_object = reflection.class_name.constantize.new
         associated_object.id = associated_object_id
-        delete_associated_object_later(associated_object)
+        delete_associated_object_later(associated_object) if associated_object.id
```

Comment:
Imagine a `user` as `main_object` with some `member_permissions`. It has `dependent: :destroy`:
https://github.com/3scale/porta/blob/762be0132ce13f3927e3ed50e39320c92020a2fe/app/models/user/permissions.rb#L7

So doing `main_object.public_send("#{reflection.name.to_s.singularize}_ids")` means `user.member_permissions_ids`, but that's overwritten:
https://github.com/3scale/porta/blob/762be0132ce13f3927e3ed50e39320c92020a2fe/app/models/user/permissions.rb#L48-L52

So `associated_object.id = associated_object_id` does nothing because it is not its real ID (it is not even an integer) and the `associated_object` does not have an ID, so doing `delete_associated_object_later(associated_object)` would raise `URI::GID::MissingModelIdError` because it cannot enqueue an object without an ID.
So this line says "if it doesn't have an ID, don't enqueue, ignore it", and it will be destroyed anyway but without parallelism... when the user is destroyed, ActiveRecord will destroy the member_permission too.

I've added a test for this scenario 😄

Reply:
Why not put the if inside the delete_associated_object_later ?

Reply:
Right, done @hallelujah 😎